vkcs/mlplatform/resource_spark_k8s.go

```diff
@@ -0,0 +1,502 @@
+				Description: "Cluster's node groups configuration",
+			},
+
+			"cluster_mode": schema.StringAttribute{
+				Required:    true,
+				Description: "Cluster Mode. Should be `DEV` or `PROD`. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf("DEV", "PROD"),
+				},
+			},
+
+			"registry_id": schema.StringAttribute{
+				Required:    true,
+				Description: "ID of the K8S registry to use with cluster. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"ip_pool": schema.StringAttribute{
+				Required:    true,
+				Description: "ID of the ip pool. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"keypair": schema.StringAttribute{
+				Optional:    true,
+				Description: "Name of the keypair. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"spark_configuration": schema.StringAttribute{
+				Optional:    true,
+				Description: "Spark configuration. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"environment_variables": schema.StringAttribute{
+				Optional:    true,
+				Description: "Environment variables. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"suspend_after_inactive_min": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Timeout of cluster inactivity before suspending. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+			},
+
+			"delete_after_inactive_min": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Timeout of cluster inactivity before deletion. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+			},
+
+			"s3_bucket_name": schema.StringAttribute{
+				Computed:    true,
+				Description: "S3 bucket name",
+			},
+
+			"history_server_url": schema.StringAttribute{
+				Computed:    true,
+				Description: "URL of the history server",
+			},
+
+			"control_instance_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "ID of the control instance",
+			},
+
+			"inactive_min": schema.Int64Attribute{
+				Computed:    true,
+				Description: "Minutes of cluster inactivity",
+			},
+
+			"region": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "The `region` to fetch availability zones from, defaults to the provider's `region`.",
```

Comment:
We usually use it to obtain client.
```suggestion
				Description: "The `region` in which ML Platform client is obtained,  defaults to the provider's `region`.",
```

Reply:
Rewrote